src/ast.rs

```diff
@@ -8,6 +8,7 @@ use std::ops::Range;
 use codespan_reporting::files::SimpleFile;
 
 use crate::constants;
+use crate::ParseErrorMsg;
```

Comment:
I don't know if it actually matters functionally, but can this be "crate::error::ParseErrorMsg" instead please?